src/prefect/cli/_prompts.py

```diff
@@ -376,7 +405,7 @@ def prompt_schedules(console) -> List[MinimalDeploymentSchedule]:
 async def prompt_select_work_pool(
     console: Console,
     prompt: str = "Which work pool would you like to deploy this flow to?",
-    client: PrefectClient = None,
+    client: Optional[PrefectClient] = None,
```

Comment:
Another thing to not block this PR on, but I added a `client_injector` that makes the typing on this kind of stuff better: https://github.com/PrefectHQ/prefect/blob/8454a1c8938bdd1204eefbdb3606129e37501653/src/prefect/client/utilities.py#L66-L74

It's used the same way as `inject_client` except that it always expects the `client` will be the first argument and non-optional: https://github.com/PrefectHQ/prefect/blob/8454a1c8938bdd1204eefbdb3606129e37501653/src/prefect/input/actions.py#L54-L57